modules/apps/document-library/document-library-preview-api/src/main/java/com/liferay/document/library/preview/background/task/BasePreviewBackgroundTaskExecutor.java

```diff
@@ -0,0 +1,135 @@
+import com.liferay.portal.kernel.backgroundtask.BaseBackgroundTaskExecutor;
+import com.liferay.portal.kernel.backgroundtask.display.BackgroundTaskDisplay;
+import com.liferay.portal.kernel.dao.orm.ActionableDynamicQuery;
+import com.liferay.portal.kernel.dao.orm.Property;
+import com.liferay.portal.kernel.dao.orm.PropertyFactoryUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.repository.model.FileVersion;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.repository.liferayrepository.model.LiferayFileEntry;
+
+import java.io.Serializable;
+
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Roberto Díaz
+ */
+public abstract class BasePreviewBackgroundTaskExecutor
+	extends BaseBackgroundTaskExecutor {
+
+	@Override
+	public BackgroundTaskExecutor clone() {
+		return this;
+	}
+
+	@Override
+	public BackgroundTaskResult execute(BackgroundTask backgroundTask)
+		throws Exception {
+
+		Map<String, Serializable> taskContextMap =
+			backgroundTask.getTaskContextMap();
+
+		long companyId = GetterUtil.getLong(taskContextMap.get("companyId"));
+
+		generatePreviews(companyId);
+
+		return BackgroundTaskResult.SUCCESS;
+	}
+
+	@Override
+	public BackgroundTaskDisplay getBackgroundTaskDisplay(
+		BackgroundTask backgroundTask) {
+
+		return null;
+	}
+
+	protected abstract void generatePreview(FileVersion fileVersion)
+		throws Exception;
+
+	protected void generatePreviews(long companyId) {
+		ActionableDynamicQuery actionableDynamicQuery =
+			dlFileEntryLocalService.getActionableDynamicQuery();
+
+		actionableDynamicQuery.setAddCriteriaMethod(
+			dynamicQuery -> {
+				Property companyIdProperty = PropertyFactoryUtil.forName(
+					"companyId");
+
+				dynamicQuery.add(companyIdProperty.eq(companyId));
+
+				Property mimeTypeProperty = PropertyFactoryUtil.forName(
+					"mimeType");
+
+				dynamicQuery.add(mimeTypeProperty.in(getMimeTypes()));
+
+				Property sizeProperty = PropertyFactoryUtil.forName("size");
+
+				dynamicQuery.add(
+					sizeProperty.le(
+						dlFileEntryConfiguration.
+							previewableProcessorMaxSize()));
+			});
+		actionableDynamicQuery.setPerformActionMethod(
+			(DLFileEntry dlFileEntry) -> {
+				FileEntry fileEntry = new LiferayFileEntry(dlFileEntry);
+
+				try {
+					generatePreview(fileEntry.getLatestFileVersion());
+				}
+				catch (Exception exception) {
+					if (_log.isWarnEnabled()) {
+						_log.warn(
+							"Unable to process file entry " +
+								fileEntry.getFileEntryId(),
+							exception);
+					}
+				}
+			});
+
+		try {
+			actionableDynamicQuery.performActions();
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException);
+		}
```

Comment:
Can you make this method throw the PortalException and log it in the `execute` method?

Reply:
Done